docs/gfx.md

```diff
@@ -0,0 +1,124 @@
+ZJS API for GFX
+===============
+
+* [Introduction](#introduction)
+* [Web IDL](#web-idl)
+* [API Documentation](#api-documentation)
+* [Sample Apps](#sample-apps)
+
+Introduction
+------------
+The GFX module provides a generic way to create pixel buffers, which can then
+be displayed on a display of some kind.  A Javascript method for initializing
+the screen, and drawing a data buffer are required to use it.
+See module/ST7735.js and samples/SPI_Screen.js for an example.
+
+Web IDL
+-------
+This IDL provides an overview of the interface; see below for documentation of
+specific API functions.
+
+```javascript
+// require returns a GFX object
+// var gfx = require('gfx');
+
+[NoInterfaceObject]
+interface GFX {
+    GFXObj init(screen width, screen height, init screen function,
+                draw function, optional this);
+};
+
+[NoInterfaceObject]
+interface GFXObj {
+    fillRect(number x coord, number y coord, number width, number height,
+             char array color);
+    drawPixel(number x coord, number y coord, char array color);
+    drawLine(number x0 coord, number y0 coord, number x1 coord,
+             number y1 coord, char array color, optional number size);
+    drawVLine(number x coord, number y coord, number height, char array color,
+              optional number size);
+    drawHLine(number x coord, number y coord, number width, char array color,
+              optional number size);
+    drawRect(number x coord, number y coord, number width, number height,
+             char array color, optional number size);
+    drawChar(number x coord, number y coord, character char, char array color,
+             optional number size);
+    drawString(number x coord, number y coord, string str, char array color,
```

Comment:
```
interface CanvasText {
  // text (see also the CanvasPathDrawingStyles and CanvasTextDrawingStyles interfaces)
  void fillText(DOMString text, unrestricted double x, unrestricted double y, optional unrestricted double maxWidth);
```

Reply:
I don't say that we need all that is in Canvas, but lets follow their style

This is currently very inconsistent:

Ie
```drawString(number x coord, number y coord, string str, char array color, optional number size);```
vs
```void fillText(DOMString text, unrestricted double x, unrestricted double y, optional unrestricted double maxWidth);```

Reply:
Color you set via

```
interface CanvasFillStrokeStyles {
  attribute (DOMString or CanvasGradient or CanvasPattern) fillStyle; // (default black)
}
```

Canvas has save() and restore() for state:

```
interface CanvasState {
  // state
  void save(); // push state on state stack
  void restore(); // pop state stack and restore state
};
```